website/content/docs/security/acl/auth-methods/aws-iam.mdx

```diff
@@ -0,0 +1,199 @@
+     to log in.
+   - If `EnableIAMEntityDetails=true`, then bound ARNs with the full path must be used, such as,
+     `arn:aws:iam::123456789012:role/path/to/MyRoleName`. Additionally, ARNs may contain a single
+     trailing wildcard. For example, `arn:aws:iam::123456789012:*` will permit any role or user in
+     the account `123456789012` to login, while `arn:aws:iam::123456789012:role/path/to/roles/*`
+     will permit only roles at the path `/path/to/roles/`.
+- `EnableIAMEntityDetails` `(bool: <false>)` - This enables the auth method to fetch the IAM role or
+  IAM user details, including tags and the full role or user path. If enabled, clients must pass the
+  `-aws-include-entity` option to `consul login`. Additionally, an IAM role or user attempting to
+  login must have `iam:GetRole` or `iam:GetUser` permission, respectively, to retrieve itself. This
+  is required in order to fetch the full path and tags of the IAM user or role, and it is required
+  in order to use wildcards in the `BoundIAMPrincipalARNs`.
+- `IAMEntityTags` `(array<string>: [])` - The list of tag keys retrieved from the IAM role or user
+  and made available to binding rules. Tags are only supported when `EnableIAMEntityDetails=true`.
+  Only the tags in this will be available in binding rules as `entity_tags.<key>`. For example, if
+  you include a `service-name` tag on an IAM role and specify the tag key in this list, then the you
+  can reference the tag value using `entity_tags.service-name` in binding rules.
+- `ServerIDHeaderValue` `(string: "")` - The value to require in the `X-Consul-IAM-ServerID` header
+  in login requests. If set, clients must include the `X-Consul-IAM-ServerID` header in the AWS API
+  requests used to login to the auth method, and the client-provided value for the header must match
+  this setting in order to successfully log in. If not set, no header is required or validated. This
+  can be used to protect against different types of replay attacks - for example, a signed request
+  sent to a dev server being resent to a production server. Consider setting this to the Consul
+  server's DNS name. When this is set, clients must set pass the `-aws-server-id-header-value`
+  option to the `consul login` command.
+- `MaxRetries` `(integer: 0)` - The maximum number of retries to use for recoverable errors when
+  making AWS API requests.
+- `IAMEndpoint` `(string: "")` - The URL where `iam:GetRole` and `iam:GetUser` requests are sent.
+  This can be used to send requests to a private endpoint or through a network proxy.
+- `STSEndpoint` `(string: "")` - The URL where `sts:GetCallerIdentity` requests are sent. This can
+  be used to send requests to a private endpoint or through a network proxy.
+- `AllowedSTSHeaderValues` `(array<string>: [])` - A list of additional allowed headers on
+  `sts:GetCallerIdentity` requests. In any case, a default list of headers AWS STS expects are
+  allowed.
+
+### Sample
+
+```json
```

Comment:
We also need to provide the HCL equivalent. Use <CodeTabs> to display both languages, with HCL being the default.

Reply:
Ah, hmm. I actually don't think this will ever by HCL. It's the [`Config`](https://www.consul.io/api-docs/acl/auth-methods#config) field in the Create Auth Method HTTP API, which requires JSON request bodies. Or, it can be passed to the CLI like, `consul acl auth-method create -config <config>`, which only supports JSON as well:

```
$ consul acl auth-method create -h
...
  -config=<string>
     The configuration for the auth method. Must be JSON. May be
     prefixed with '@' to indicate that the value is a file path to load
     the config from. '-' may also be given to indicate that the config
     is available on stdin
...
```

I don't think this can go into an HCL file for consul agent config, iirc.

Reply:
Gotcha, I am following now. Thanks for clarifying Paul.